discord.go

```diff
@@ -26,6 +26,17 @@ const VERSION = "0.19.0"
 // ErrMFA will be risen by New when the user has 2FA.
 var ErrMFA = errors.New("account has 2FA enabled")
 
+// Auth represents authentication options to use with new.
+type Auth struct {
```

Comment:
I like the idea of this being a PasswordAuth struct, and then having 2 fields. Then have a TokenAuth struct with just one field.

No reason to overload the two methods of authentication again, it was confusing as arguments, I don't want it to be confusing with struct members.

Reply:
I just thought it should have all three, since in the docs for `New` it says
```
With an email, password and auth token - Discord will verify the auth
token, if it is invalid it will sign in with the provided
credentials. This is the Discord recommended way to sign in.
```
So i thought it would be bad to remove the option of using both.